lib/usd/ui/layerEditor/layerTreeItem.h

```diff
@@ -96,12 +99,16 @@ class LayerTreeItem : public QStandardItem
     bool isMuted() const;
     // check if this layer is muted, or any of its parent
     bool appearsMuted() const;
+    // check if this layer is readonly
+    bool isReadOnly() const;
     // true if dirty, but look at needsSaving for UI feedback
-    bool isDirty() const { return _layer ? _layer->IsDirty() : false; }
+    bool isDirty() const { return _layer ? _layer->IsDirty() && !isReadOnly() : false; }
```

Comment:
You've modified the behavior of this method. It is no longer "isDirty" Now it is "isDirty and not ReadOnly". Why? I think the code should have to call both these methods (isDirty/isReadOnly) if that is the answer they want.

Reply:
Yea your right!